invesalius/gui/task_navigator.py

```diff
@@ -1028,9 +1033,29 @@ def OnImageFiducials(self, n, evt):
             Publisher.sendMessage('Set image fiducial', fiducial_name=fiducial_name, coord=np.nan)
             Publisher.sendMessage('Delete fiducial marker', marker_id=marker_id)
 
-    def OnTrackerFiducials(self, n, evt):
-        fiducial_name = const.TRACKER_FIDUCIALS[n]['fiducial_name']
-        Publisher.sendMessage('Set tracker fiducial', fiducial_name=fiducial_name)
+    def OnTrackerFiducials(self, n, evt, ctrl):
+
+        # Do not allow several tracker fiducials to be set at the same time.
+        if self.tracker_fiducial_being_set is not None and self.tracker_fiducial_being_set != n:
+            ctrl.SetValue(False)
+            return
+
+        # Called when the button for setting the tracker fiducial is enabled and either pedal is pressed
+        # or the button is pressed again.
+        #
+        def set_fiducial_callback():
+            fiducial_name = const.TRACKER_FIDUCIALS[n]['fiducial_name']
+            Publisher.sendMessage('Set tracker fiducial', fiducial_name=fiducial_name)
+            self.pedal_connection.remove_callback('fiducial')
+
+            ctrl.SetValue(False)
+            self.tracker_fiducial_being_set = None
+
+        if ctrl.GetValue() is True:
```

Comment:
It's not wrong to check if something is true this way, but it's better to use:

```
if ctrl.GetValue():
```

or 

```
if ctrl.GetValue() == True:
```

The first way is better. `is` is used to test if two objects are the same. It works with `True` because Python creates only one `True` object, so every `True` is the same.

Reply:
Thanks for the comment!

On the page

https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html

it actually seems to say that `if flag == True:` is another non-preferred way, whereas `if flag is True:` is preferred over it, but the PEP 8 preference is just `if flag:`, as you also proposed.

My reason for being explicit about the True value here was that when reading the code, I'm sometimes confused by some of these boolean checks -- I'm not sure if they actually check the value being True or False or the value being None or not None. That's why I wanted to make it clear here. (Without seeing the context, it could be that we just want to know that GetValue() returns a value that is not None, that is, the value has been set.)

But I'm happy to go with the PEP 8 preference, so I changed it the way you proposed.